contentlayer.config.ts

```diff
@@ -1,18 +1,116 @@
-import { defineDocumentType, makeSource, ComputedFields } from "contentlayer2/source-files";
-
-// this is a basic example to show how to structure your content source
-export const _DataName = defineDocumentType(() => ({
-  name: "Post", // name of the post
-  filePathPattern: `**/*.md`, // file path where the mdx file to be process is located
-  contentType: "mdx", // content type mdx
-  // fields: types desc
+import * as fs from "fs";
+import { defineDocumentType, defineNestedType, makeSource } from "contentlayer2/source-files";
+
+const Resources = defineNestedType(() => ({
+  name: "Resources",
   fields: {
-    title: { type: "string", required: true },
-    date: { type: "date", required: true },
-  },
-  computedFields: {
-    url: { type: "string", resolve: (post) => `/posts/${post._raw.flattenedPath}` },
+    title: { type: "string" },
+    url: { type: "string" },
   },
 }));
 
-export default makeSource({ contentDirPath: "posts", documentTypes: [_DataName] });
+const path = `${process.cwd()}/public/bitcoin-transcript`;
+
+const generateDocuTypes = () => {
+  const getFolders = fs.readdirSync(path, "utf-8");
+
+  // getFolders is sliced @6 to remove files which are not folders like .gitignore, readme.md , .github etc. We only want to read files from the folders that contain transcripts.
+  const folders = getFolders.slice(6).filter((item) => item !== "twitter_handles.json");
+
+  // firstLetterRegex: ** /(^\w{1})|(\s+\w{1})/g ** gets the first letter of the split folder name and capitalizes the first letter of the each word. This is to create a camelCase naming Convention for the generated folders
+  const firstLetterRegex = /(^\w{1})|(\s+\w{1})/g;
+
+  const DefinedDocumentTypes = folders.map((name) => {
+    const slugifyName = name
+      .split("-")
+      .map((text) => text.replace(firstLetterRegex, (text) => text.toUpperCase()))
+      .join("");
+
+    return defineDocumentType(() => ({
+      name: slugifyName,
```

Comment:
You are defining a new document type for each folder. Why is that? All those are transcripts, why not defining everything as `Transcript`? in the same way that in the example they are defined as `Posts`

Reply:
You have to define a document type for each folder you want contentlayer to read files from